mkosi/sandbox.py

```diff
@@ -5,18 +5,45 @@
 import uuid
 from collections.abc import Sequence
 from pathlib import Path
-from typing import Optional, Protocol
+from typing import NamedTuple, Optional, Protocol
 
 from mkosi.types import PathString
 from mkosi.user import INVOKING_USER
 from mkosi.util import flatten, one_zero, startswith
 
 
+class Mount(NamedTuple):
+    src: PathString
+    dst: PathString
+    devices: bool = False
+    ro: bool = False
+    required: bool = True
+
+    def __hash__(self) -> int:
+        return hash((Path(self.src), Path(self.dst), self.devices, self.ro, self.required))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Mount):
+            return False
+
+        return self.__hash__() == other.__hash__()
```

Comment:
Why not make this a dataclass and get this for free?

Reply:
I'm not sure if the dataclass `__eq__` would do what we need. 

> eq: If true (the default), an [__eq__()](https://docs.python.org/3/reference/datamodel.html#object.__eq__) method will be generated. This method compares the class as if it were a tuple of its fields, in order. Both instances in the comparison must be of the identical type.

We don't want a direct comparison, we want a comparison after converting the `PathString` fields to `Path`